components/admin/lessons/AdminLessonInputs.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react'
+import { useAddModuleMutation } from '../../../graphql'
+import { formChange } from '../../../helpers/formChange'
+import { FormCard, Option } from '../../FormCard'
+import { AlertFillIcon, CheckCircleIcon } from '@primer/octicons-react'
+import styles from '../../../scss/adminLessonInputs.module.scss'
+import { Spinner } from 'react-bootstrap'
+import { GraphQLError } from 'graphql-request/dist/types'
+
+type Props = {
+  lesson: { title: string; id: number }
+}
+
+enum Error {
+  INVALID_DATA = 'missing module name or description'
+}
+
+const values: Option[] = [
+  {
+    title: 'Module Name'
+  },
+  {
+    title: 'Description',
+    type: 'MD_INPUT'
+  }
+]
+
+const AdminLessonInputs = ({ lesson }: Props) => {
+  const [formOptions, setFormOptions] = useState(values)
+  const [name, content] = formOptions
+
+  const [addModuleMutation, { data, error, loading }] = useAddModuleMutation({
+    variables: {
+      content: (content.value as string) || '',
+      lessonId: lesson.id,
+      name: (name.value as string) || ''
+    }
+  })
+
+  const [errorMsg, setErrorMsg] = useState(error?.message || '')
+
+  const handleChange = async (value: string, propertyIndex: number) => {
+    await formChange(value, propertyIndex, formOptions, setFormOptions)
+  }
+
+  const OnSubmit = async () => {
+    try {
+      // Empty strings aren't handled in the resolver
+      if (!content.value || !name.value) {
+        return setErrorMsg(Error.INVALID_DATA)
+      }
+
+      setErrorMsg('')
+      await addModuleMutation()
+    } catch (err) {
+      setErrorMsg((err as GraphQLError).message)
+    }
+  }
+
+  const Message = () => {
+    if (loading) {
+      return (
+        <div className={styles.loading}>
+          <Spinner animation="grow" size="sm" />
+          <span>Adding the module...</span>
+        </div>
+      )
+    }
+
+    if (errorMsg || error) {
+      return (
+        <div className={styles.error}>
+          <AlertFillIcon />
+          <span>Failed to add the module: {error?.message || errorMsg}</span>
+        </div>
+      )
+    }
+
+    if (data) {
+      return (
+        <div className={styles.success}>
+          <CheckCircleIcon />
+          <span>
+            Added the module <strong>{data.addModule.name}</strong>{' '}
```

Comment:
Why is `{' '}` needed?

Reply:
It's auto-added by the formatter, but I think it's safe to remove it.

Reply:
Yeah lets remove it if its not needed

Reply:
Also, is it guaranteed that `data.addModule` and `data.addModule.name` both exist? If not guaranteed, then this could lead to an issue due to property not existing

Reply:
If there's `data`, there's `data.addModule`. But I'm not sure about `data.addModule.name`. If the property name changes, an issue will appear BUT, if that ever happens, TypeScript will scream with `name does not exist in { ... }`.

Reply:
Is `name` something that is guaranteed to be a field of the `addModule` object? If the `name` isn't guaranteed to exist, then `addModule.name` will output `undefined`. I'm worried that will result in the word `undefined` being rendered here

Reply:
Doesn't the [query](https://github.com/flacial/c0d3-app/blob/673ac51e43be35c7a3e125a26cfd0060a5104d83/graphql/queries/addModule.ts#L7) definition guarantee that the field will always exist?

Reply:
The query definition guarantee that the field will always exists, but it does not guarantee it will always be non-null, because fields can return `null`. 

I just looked up and confirmed that `name` will always return a non-null value though.
For the `addModule` resolver, the typedefs said it will return `Module`, and it will always be non-null
https://github.com/garageScript/c0d3-app/blob/master/graphql/typeDefs.ts#L59

And then in `Module` typedef, we have a `name` field that is also non-nullable and must be a string.
https://github.com/garageScript/c0d3-app/blob/master/graphql/typeDefs.ts#L241

If `name` was not non-nullable that might have been problematic edge case since the word `null` might be rendered when `name` is null.

Reply:
> Why is {' '} needed?

@anthonykhoa When a line length is greater than the max length, it has to be wrapped to the next line. In JSX's case, it needs this `{'  '}` or there won't be a space between the text.